firebase-messaging/src/main/java/com/google/firebase/messaging/FirebaseMessagingService.java

```diff
@@ -232,7 +249,7 @@ private void dispatchMessage(Intent intent) {
         MessagingAnalytics.logNotificationForeground(intent);
```

Comment:
at the moment the change will bypass this `MessagingAnalytics` call, but I think that's probably okay?

if someone uses this new API to bypass default notification logic, the implication is that it should be treated identically to a standard data message (including for analytics)